### ISSUE TITLE
x s null on at least one execution path.

### ISSUE THREAD

**Issue by thomhurst** [SonarSource/sonar-dotnet#7995]
I'm getting this on a Codacy report, which I think internally uses Sonar.

You can literally see from the code snipped provided that I'm checking for not null or empty. Yet I get a null warning.

```csharp
if (!string.IsNullOrWhiteSpace(secret))
{
    yield return secret;
}
```

`'secret' is null on at least one execution path.`

<img width="378" alt="image" src="https://github.com/SonarSource/sonar-dotnet/assets/30480171/b2fe80a3-f27d-4dbc-9529-58d281855f94">

**Comment by gregory-paidis-sonarsource** [comment]
Hey @thomhurst.

I tried this locally, at the latest version of our analyzer, and the rule is not raised.
Here is the snippet I used:

```csharp
private static IEnumerable<string> Method(string secret)
{
    if (!string.IsNullOrWhiteSpace(secret))
    {
        yield return secret;
    }
}
```

There are two possible problems here:

- Codacy is packaging `SonarC#` is a different way than we do, so some precision might be lost
- Codady's version of `SonarC#` is lagging a bit behind our latest releases.

Unfortunately, we cannot do much about any of these, since we do not have control over when they update our analyzer, as well as how they package it.

If you are curious about how behind `SonarC#` is, you could check what version of `sonar-dotnet` is used by Codacy, and compare it with our latest release.

Also, if I missed something and you can reproduce this in a sonarcloud public repo and give me a link, I would be happy to look into it further.
